### PR TITLE
boot: Add PC speaker support

### PR DESCRIPTION
**Pull request by medhefgo**
Fixes: #17508

### CODE REVIEW

File: src/boot/efi/boot.c [hunk 1]
```diff
@@ -725,6 +727,11 @@ static BOOLEAN menu_run(
                         ST->ConOut->OutputString(ST->ConOut, clearline + 1 + x + len);
                 }
 
+                if(do_beep) {
```
Comment (bluca) [inline src/boot/efi/boot.c]: missing space

**Review by bluca — APPROVED** [review]
hackaton project: play the windows 95 startup theme when the windows bootloader is selected

**Review by alex19EP — APPROVED** [review]
tested it in qemu. all works fine.

**Comment by sthibaul** [comment]
> > hackaton project: play the windows 95 startup theme when the windows bootloader is selected
> 
> actually we can further improve convenience for the blind by adding the ability to add a different number of beeps to the boot-entry.

Yes, some people find it useful, I had introduced the `play` command in grub to that end, and people do use it to be able to distinguish between boot entries while moving blindly.

**Comment by medhefgo** [comment]
> Yes, some people find it useful, I had introduced the `play` command in grub to that end, and people do use it to be able to distinguish between boot entries while moving blindly.

Would it be enough to beep n times when selecting the nth entry? (I'd rather not make an explosion of config (entry) options unless it's truly needed. If anyone wants full control, they really ought to use grub.)

**Comment by sthibaul** [comment]
> Would it be enough to beep n times when selecting the nth entry?

That would already be plenty, yes!

Thanks a lot for your work on this!

**Comment by medhefgo** [comment]
See https://github.com/systemd/systemd/pull/22140. Let me know if the beep frequency is too fast or slow.

**Pull request merged**